### PR TITLE
Slot Side Option Complete

### PR DESCRIPTION
**Pull request by laz1861**
It took a re-write of the way the code generated the tabs to make it compatible with generating slots.  I've tested this visually with all cases. I've constructed one box and it's working.   This incorporates my earlier change of making sides be groups. There were some things that were removed:
The fixed tab width option was removed.  Tabs and gaps are now always the same size.
The alternating tabs layout was removed.  The boxes are all symmetric now.

There are a few things that may want additional fixing:
The program has always made the tabs too short by the kerf amount.  It ends up with tabs being slightly recessed relative to its matching surface.  We can fix that to make sure that tabs are flush.
The slots are all currently objects, not paths.  This requires a manual change from object to path.  Ideally, they would all be entered as paths.

### CODE REVIEW

File: extensions/kmlaser_boxmaker.py [hunk 1]
```diff
@@ -33,76 +33,734 @@
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 '''
-__version__ = "0.8" ### please report bugs, suggestions etc to bugs@twot.eu ###
+__version__ = "0.3" ### please report bugs, suggestions etc to board@knoxmakers.org ###
```
Comment (bmanifold) [inline extensions/kmlaser_boxmaker.py]: Out of curiosity, is there a reason for dropping this number from 0.8 to 0.3 ?

Comment (laz1861) [inline extensions/kmlaser_boxmaker.py]: That was someone else's version number, and I basically rewrote the whole
engine.  I started out at 0.1 in my work, and this ended up at 0.3 before
the pull request.  Is that bad form ?

Comment (bmanifold) [inline extensions/kmlaser_boxmaker.py]: Well, if you rewrote the entire engine, then I understand starting from 0.1

The thing that is getting confusing is the versioning between individual components and the overall KM-Laser package.  We should probably try to figure out how we want to version everything.

File: extensions/kmlaser_boxmaker.py [hunk 2]
```diff
@@ -33,76 +33,734 @@
...
+      gapcount = (int(divs)-1)/2
+      for i in range(0,gapcount):
+        #create a rectangle at each of gaps
+        #no special widths because tabs are on the ends
+        drawR((Vx,Vy),(gapWidth,thickness),grp)
+        #after every rectangle, move +x
+        #move amount: gapwidth + tabwith
+        Vx+=gapWidth + tabWidth
+        #increment loop counter
+        i+=1
```
Comment (bmanifold) [inline extensions/kmlaser_boxmaker.py]: There is no need to increment the variable i here
The for loop increments i on it's own.

This same comment can be applied to all of the for loops that increment i

Comment (laz1861) [inline extensions/kmlaser_boxmaker.py]: These are good comments.  It really makes me miss our old programming
nights.  I learned a ton of stuff doing that.  I'd love to set up a time
where we can sit down and go through the things that you'd recommend
changing.  I have a feeling I could learn a lot that way.

**Pull request merged**